### DISCUSSION TITLE
Conditional globalSearch

### DISCUSSION THREAD

**Discussion by aotearoait** [filamentphp/filament#12886] [Ideas]
Most settings in the panel are able to be conditionally set, eg ->tenantMenu(auth()->user()->isAdmin())
But the global search does not accept closures.
I cannot find a way to do with middleware either as it just gets overriden.
So the request is to allow closures for globalSearch()

**Comment by dellow** [comment]
I agree with this. Would like to disable global search for some users and not others.

**Comment by tiagof** [comment]
```php
class MyPanelProvider extends PanelProvider
{
    public function panel(Panel $panel): Panel
    {
        $panel->globalSearch(true/false);
...
```

**Reply by dellow** [reply]
This doesn't solve the issue?
